Fix to handle weird scenarios where the request urls to assets contain query string

infindex opened the pull request.

infindex committed:
Fix to handle weird scenarios where the request urls contain query
parameters

jrgm commented:
So, for one thing, the tests no longer pass. Also a description of the problem you are trying to solve would help.

ozten commented on lib/connect-cachify.js:
```diff
@@ -9,7 +9,8 @@ var crypto = require('crypto'),
     path = require('path'),
-    und = require('underscore');
+    und = require('underscore'),
+    URL = require('url');
```
variable holding the module should be lowercase.

ozten commented on lib/connect-cachify.js:
```diff
@@ -114,7 +117,7 @@ exports.setup = function (assets, options) {
         if (exists === true && hash === actualHash) {
           resp.setHeader('Cache-Control', 'public, max-age=31536000');
-          req.url = url;
+          req.url = url + '?' + req_url.query; // append back the query string
```
Awesome! Can you add a test that demonstrates the problem that query string parameters introduce?

infindex committed:
Merge remote-tracking branch 'upstream/master'

infindex committed:
- Fixed test failures
- Handle query parameter in development mode
- Adding tests to demonstrates the problem when query string parameters
  Are introduced

infindex commented:
@jrgm  - fixed the test failures...
@ozten - additional test cases added to demonstrate the query string problem. also fixed the variable name as suggested.

Also, added query string parameter handling in development mode and urls with Hash ids.

Thanks!
MG

ozten commented on test/connect-cachify-test.js:
```diff
@@ -261,6 +334,32 @@ exports.setup = nodeunit.testCase({
+        root: '/tmp'
+      });
+    var link = cachify.cachify_js("/other", {hash: 'd41d8cd98f'});
+    test.equal(link, '<script src="/cachify/d41d8cd98f/other"></script>');
```
Awesome, thanks for adding all these tests.
